test/gcs/container_test.go

```diff
@@ -0,0 +1,240 @@
+//go:build linux
+
+package gcs
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/oci"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/Microsoft/hcsshim/internal/guest/gcserr"
+	"github.com/Microsoft/hcsshim/internal/guest/stdio"
+
+	testoci "github.com/Microsoft/hcsshim/test/internal/oci"
+)
+
+//
+// tests for operations on standalone containers
+//
+
+// todo: using `oci.WithTTY` for IO tests is broken and hangs
+
+func TestContainerCreate(t *testing.T) {
+	requireFeatures(t, featureStandalone)
+
+	ctx := context.Background()
+	host, rtime := getTestState(ctx, t)
+	assertNumberContainers(ctx, t, rtime, 0)
+
+	id := t.Name()
+	c := createStandaloneContainer(ctx, t, host, id)
+	t.Cleanup(func() {
+		cleanupContainer(ctx, t, host, c)
+	})
+
+	p := startContainer(ctx, t, c, stdio.ConnectionSettings{})
+	t.Cleanup(func() {
+		killContainer(ctx, t, c)
+		waitContainer(ctx, t, c, p, true)
+	})
+
+	assertNumberContainers(ctx, t, rtime, 1)
+	css := listContainerStates(ctx, t, rtime)
+	cs := css[0]
```

Comment:
technically `assertNumberContainers` calls `listContainerStates` and checks the length is 1, but I'll combine the two, so that `assertNumberContainers` returns the container stats from `listContainerStates` if that many are found